test/unit/StakingProducts/recalculateEffectiveWeight.js

```diff
@@ -0,0 +1,98 @@
+const { ethers } = require('hardhat');
```

Comment:
imo we can have unit tests for 

getEffectiveWeight

by itself. it's a publicly exposed function.